Sources/SmokeDynamoDB/DynamoDBTableHistoricalItemExtensions.swift

```diff
@@ -274,40 +276,41 @@ public extension DynamoDBTable {
                                                                              withRetries: retries - 1,
                                                                              completion: completion)
                 } catch {
-                    completion(error)
+                    completion( HTTPResult.error( error ) )
                 }
-            } else {
-                completion(error)
+            } else if let theError = error {
+                completion( HTTPResult.error( theError ) )
             }
         }
-        
+
         func handleGetItemResult(result: HTTPResult<TypedDatabaseItem<AttributesType, ItemType>?>) {
             switch result {
             case .response(let existingItemOptional):
                 guard let existingItem = existingItemOptional else {
                     let error = SmokeDynamoDBError.conditionalCheckFailed(paritionKey: compositePrimaryKey.partitionKey,
                                                               sortKey: compositePrimaryKey.sortKey,
                                                               message: "Item not present in database.")
-                    
-                    return completion(error)
+
+                    return completion( HTTPResult.error( error ) )
                 }
-                
+
                 do {
                     let updatedItem = try primaryItemProvider(existingItem)
                     let historicalItem = historicalItemProvider(updatedItem)
-                    
+
                     try updateItemWithHistoricalRowAsync(primaryItem: updatedItem,
                                                          existingItem: existingItem,
                                                          historicalItem: historicalItem,
                                                          completion: handleUpdateItemResult)
+                    completion( HTTPResult.response( updatedItem ) )
```

Comment:
`updateItemWithHistoricalRowAsync` is already being passed a completion handler - `handleUpdateItemResult` - that it will complete when it succeeds or fails. So adding this line here will execute the completion handler with a success and then (or before due to threading) `updateItemWithHistoricalRowAsync` will execute the completion handler with an error or no error.

What we want to do is pass `updateItemWithHistoricalRowAsync` a modified `handleUpdateItemResult` completion handler that will call `completion` with the updatedItem on success (no error) or propagate the error on failure.

Reply:
This opened up a whole can of worms, and I ended up changing all of use cases of the `Error?` completion handlers to `HTTPResult` ones.

Reply:
I don't think we need to go down this rabbit role. I was more thinking of something like this-

https://github.com/amzn/smoke-dynamodb/commit/91c0f5f0d6abdf0d2e316cfd3258ba4fe9e3fcc3

Reply:
Got it, there was a disconnect on my part because I thought we should replace all of the completion handlers to handle `HTTPResult`. Thanks for the example, I've updated the branch with your suggestion.